<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>676</width>
    <height>416</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Unicycle Controller</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="PID_gains_QL">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>40</y>
      <width>171</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Noto Sans Adlam</family>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Set PID gains</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="PHIKP_QDSB">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>70</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="value">
     <double>25.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="PHIKP_QL">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>100</y>
      <width>61</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PHIKP</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="PHIKP_OLD_QDSB">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>70</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="PHIKP_OLD_QL">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>100</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PHIKP_OLD</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="PHIKI_QDSB">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>130</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="value">
     <double>5.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="PHIKI_OLD_QL">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>160</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PHIKI_OLD</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="PHIKI_OLD_QDSB">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>130</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="PHIKI_QL">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>160</y>
      <width>61</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PHIKI</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="PHIKD_QDSB">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>190</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="value">
     <double>0.200000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="PHIKD_OLD_QL">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>220</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PHIKD_OLD</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="PHIKD_OLD_QDSB">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>190</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="PHIKD_QL">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>220</y>
      <width>61</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PHIKD</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Update_PID_QPB">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>290</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Update PID</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Revert_PID_QPB">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>290</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Revert PID</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="XTRAVEL_QDSB">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>70</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="value">
     <double>300.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="XTRAVEL_QL">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>100</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>XTRAVEL</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="TURN_ANGLE_QDSB">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>130</y>
      <width>62</width>
      <height>26</height>
     </rect>
    </property>
    <property name="maximum">
     <double>360.000000000000000</double>
    </property>
    <property name="value">
     <double>45.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="TURN_ANGLE_QL">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>160</y>
      <width>91</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>TURN_ANGLE</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Forward_QPB">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>80</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Forward</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Right_QPB">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>110</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Right</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Left_QPB">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>110</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Left</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Back_QPB">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>140</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Back</string>
    </property>
   </widget>
   <widget class="QLabel" name="Log_QL">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>200</y>
      <width>171</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Noto Sans Adlam</family>
      <pointsize>14</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Log:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="Log_QTE">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>230</y>
      <width>301</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>676</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
